### PR TITLE
fix: add better keys and fuller values to influxDB analytics

### PR DESCRIPTION
**Pull request by SMillerDev**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
This resolves most of the issues I've talked about with @MikeMcQuaid. After this I'll drop the `install` analytics and let it populate again.

### CODE REVIEW

File: Library/Homebrew/utils/analytics.rb [hunk 1]
```diff
@@ -306,14 +306,15 @@ def clear_os_arch_prefix_ci
...
-            developer: Homebrew::EnvConfig.developer?,
-            version:   HOMEBREW_VERSION,
-            system:    HOMEBREW_SYSTEM,
-            ci:        ENV["CI"].present?,
-            arch:      arch_label(verbose: verbose),
-            prefix:    custom_prefix_label(verbose: verbose),
+            version:             HOMEBREW_VERSION.match(/^([\d\.]*)-?/)[1],
+            prefix:              custom_prefix_label(verbose: verbose),
+            ci:                  ENV["CI"].present?,
+            developer:           Homebrew::EnvConfig.developer?,
```
[the remark this answers is not in the document]
Comment (SMillerDev) [reply Library/Homebrew/utils/analytics.rb:312]: what's the difference?

[the remark this answers is not in the document]
Comment (MikeMcQuaid) [reply Library/Homebrew/utils/analytics.rb:327]: `developer?` is set manually, the other indicates they are in the "beta channel"

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Library/Homebrew/utils/analytics.rb:325]: I think developer will the closest fit to "is this a developer install", if it is not we should probably expand what `Homebrew::EnvConfig.developer?` means.

File: Library/Homebrew/utils/analytics.rb [hunk 2]
```diff
@@ -290,10 +291,21 @@ def custom_prefix_label(verbose: false)
...
+          "x86_64"
         else
           ""
         end
       end
+      alias generic_arch_label arch_label
+
+      sig { returns(String) }
+      def homebrew_version
+        version = HOMEBREW_VERSION.match(/^([\d.]*)-?/)[1]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/analytics.rb:304]: @SMillerDev What's this look like for normal versions vs. developer versions?

File: Library/Homebrew/utils/analytics.rb [hunk 3]
```diff
@@ -305,17 +317,20 @@ def clear_os_arch_prefix_ci
...
+            arch:                arch_label(verbose: verbose),
+            os:                  HOMEBREW_SYSTEM,
+            os_name_and_version: OS_VERSION,
           }
+
           unless verbose
             data.delete(:arch) if data[:arch].blank?
-            data.delete(:prefix) if Homebrew.default_prefix?
+            data.delete(:google_prefix) if Homebrew.default_prefix?
+            data[:prefix] = :default if Homebrew.default_prefix?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/analytics.rb:333]: Combine these `if`s?.

Also: we want the opposite: only include this if it's a default prefix so we don't leak e.g. usernames from home directories when it's a non-default prefix. For all non-default prefixes we can just use "non-default prefix".

File: Library/Homebrew/utils/analytics.rb [hunk 4]
```diff
@@ -150,7 +151,7 @@ def report_build_error(exception)
         if (options = exception.options.to_a.map(&:to_s).join(" ").presence)
           action = "#{action} #{options}".strip
         end
-        report_event("BuildError", action)
+        report_event(:build_error, action)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/analytics.rb:154]: Probably want to make sure this stays as `BuildError` for Google.

File: Library/Homebrew/utils/analytics.rb (hunk 2 above)
Comment (SMillerDev) [reply Library/Homebrew/utils/analytics.rb:304]: It's either 4.0.0 or 4.0.0-dev

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**